source/keyring_trace.c

```diff
@@ -20,18 +20,17 @@ bool aws_cryptosdk_keyring_trace_is_valid(const struct aws_array_list *trace) {
         return false;
     }
     AWS_FATAL_PRECONDITION(trace->item_size == sizeof(struct aws_cryptosdk_keyring_trace_record));
+    if (!aws_array_list_is_valid(trace)) {
+        return false;
+    }
     /* iterate over each record in the list */
-    size_t num_records = aws_array_list_length(trace);
-    bool is_valid      = aws_array_list_is_valid(trace);
-    for (size_t idx = 0; idx < num_records; ++idx) {
-        struct aws_cryptosdk_keyring_trace_record *record;
-        if (!aws_array_list_get_at_ptr(trace, (void **)&record, idx)) {
-            /* check if each record is valid */
-            bool record_is_valid = aws_cryptosdk_keyring_trace_record_is_valid(record);
-            is_valid &= record_is_valid;
```

Comment:
This was not working well for me neither so I am going to use it in my proofs too. Thanks!

Reply:
Sure! For me, it's also that stubbing out `aws_array_list_get_at_ptr` improves performance quite a bit, so it made sense to replace it here. There are a number of other proofs even outside of the Hensel work using that stub as well.